STYLE_GUIDE.md

```diff
@@ -0,0 +1,394 @@
+# 1. Style and Conventions
+
+## 1.1 Style Guide
+
+Streaming generally follows Google's
+[Python Style Guide](https://google.github.io/styleguide/pyguide.html) for how to format and structure code.
+
+## 1.2. Pre-Commit Hooks
+
+Streaming uses [Pre Commit](https://pre-commit.com/) to enforce style checks. To configure, run
+```
+pip install '.[dev]'  # if not already installed
```

Comment:
[nit] Not sure this applies here but on Composer we had to switch to [all] bc Pyright needs all packages to perform checks.

Reply:
Not required since our linting uses `.[dev]`. For example, https://github.com/mosaicml/streaming/blob/main/.github/workflows/linting.yaml#L26